Lib/typing.py

```diff
@@ -930,6 +968,21 @@ def __subclasscheck__(self, cls):
                 return True
 
 
+def _value_and_type_iter(parameters):
+    return ((p, type(p)) for p in parameters)
+
+
+class _LiteralGenericAlias(_GenericAlias, _root=True):
+
+    def __eq__(self, other):
+        if not isinstance(other, _LiteralGenericAlias):
+            return NotImplemented
+
+        return set(_value_and_type_iter(self.__args__)) == set(_value_and_type_iter(other.__args__))
```

Comment:
I think we can skip this because PEP said that `Literal` can be used with mutable values, we don't raise an exception when creating `Literal` with mutable values but I think compare can raise an exception.

What do you think about that?

Reply:
That's an interesting point. For the most part, `Literal` should be used by static type checkers and not in equality tests, so it makes sense. I'm just afraid someone gets this error in the future and files a bug report.

Reply:
As an option, we can catch a `TypeError` and raise a new `TypeErrror` with information about the usage of mutable value.